leader-election-impl/src/test/java/com/palantir/paxos/PaxosLatestRoundVerifierTest.java

```diff
@@ -0,0 +1,84 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.paxos;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.List;
+import java.util.concurrent.Executors;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+
+// TODO(nziebart): these tests are mostly sanity checks, until we have better tests for {@link PaxosQuorumChecker}.
+public class PaxosLatestRoundVerifierTest {
+
+    private static final long ROUND = 5L;
+    private static final long LATER_ROUND = 6L;
+
+    private final PaxosAcceptor acceptor1 = mock(PaxosAcceptor.class);
+    private final PaxosAcceptor acceptor2 = mock(PaxosAcceptor.class);
+    private final PaxosAcceptor acceptor3 = mock(PaxosAcceptor.class);
+
+    private final List<PaxosAcceptor> acceptors = ImmutableList.of(
+            acceptor1,
+            acceptor2,
+            acceptor3);
+
+    private final PaxosLatestRoundVerifierImpl verifier = new PaxosLatestRoundVerifierImpl(acceptors, 2,
+            Executors.newCachedThreadPool());
+
+    @Test
+    public void hasQuorumIfAllNodesAgree() {
+        when(acceptor1.getLatestSequencePreparedOrAccepted()).thenReturn(ROUND);
+        when(acceptor2.getLatestSequencePreparedOrAccepted()).thenReturn(ROUND);
+        when(acceptor3.getLatestSequencePreparedOrAccepted()).thenReturn(ROUND);
+
+        assertThat(verifier.isLatestRound(ROUND)).isEqualTo(PaxosQuorumStatus.QUORUM_AGREED);
+    }
+
+    @Test
+    public void hasQuorumIfQuorumAgrees() {
+        when(acceptor1.getLatestSequencePreparedOrAccepted()).thenReturn(ROUND);
+        when(acceptor2.getLatestSequencePreparedOrAccepted()).thenReturn(ROUND);
+        when(acceptor3.getLatestSequencePreparedOrAccepted()).thenReturn(LATER_ROUND);
+
+        assertThat(verifier.isLatestRound(ROUND)).isEqualTo(PaxosQuorumStatus.QUORUM_AGREED);
+    }
+
+    @Test
+    public void doesNotHaveQuorumIfQuorumFails() {
+        when(acceptor1.getLatestSequencePreparedOrAccepted()).thenThrow(new RuntimeException("foo"));
+        when(acceptor2.getLatestSequencePreparedOrAccepted()).thenThrow(new RuntimeException("foo"));
+        when(acceptor3.getLatestSequencePreparedOrAccepted()).thenReturn(ROUND);
+
+        assertThat(verifier.isLatestRound(ROUND)).isEqualTo(PaxosQuorumStatus.NO_QUORUM);
+    }
+
+    @Test
+    public void hasDisagreementIfQuorumDoesNotAgree() {
+        when(acceptor1.getLatestSequencePreparedOrAccepted()).thenReturn(ROUND);
+        when(acceptor2.getLatestSequencePreparedOrAccepted()).thenReturn(LATER_ROUND);
+        when(acceptor3.getLatestSequencePreparedOrAccepted()).thenReturn(LATER_ROUND);
+
+        assertThat(verifier.isLatestRound(ROUND)).isEqualTo(PaxosQuorumStatus.SOME_DISAGREED);
+    }
+
```

Comment:
nit: Might want a check that a smaller round is considered accepted as well (since that one is part of the function which is specific to `PaxosLatestRoundVerifierImpl`).

Reply:
hmm, not sure i understand this. can you explain more?

Reply:
Yep. I meant something like this:

```java
when(acceptor1.getLatestSequencePreparedOrAccepted()).thenReturn(ZERO);
when(acceptor2.getLatestSequencePreparedOrAccepted()).thenReturn(ROUND);
when(acceptor3.getLatestSequencePreparedOrAccepted()).thenReturn(LATER_ROUND);
assertThat(verifier.isLatestRound(ROUND)).isEqualTo(PaxosQuorumStatus.QUORUM_AGREED);
```

In other words, this is a check that verifies we treat earlier rounds as acks. I saw the comment on better tests for `PaxosQuorumChecker`, but this one is really a piece of `PaxosLatestRoundVerifier.acceptorAgreesIsLatestRound`.